### PR TITLE
Clean up compiler warnings

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #71 into master will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/blackducksoftware/blackduck-jira/pull-requests/71)

**Review by jamesrichard91 — APPROVED** [review]

File: src/main/java/com/blackducksoftware/integration/jira/config/JiraConfigDeserializer.java [hunk 1]
```diff
@@ -44,14 +47,23 @@ public BlackDuckJiraConfigSerializable deserializeConfig(final PluginIssueCreati
...
         config.setPolicyRulesJson(ticketCriteria.getPolicyRulesJson());
+
         logger.debug("Mappings:");
-        for (final BlackDuckProjectMapping mapping : config.getHubProjectMappings()) {
-            logger.debug(mapping.toString());
+        final Set<BlackDuckProjectMapping> projectMappings = config.getHubProjectMappings();
+        if (null != projectMappings) {
+            for (final BlackDuckProjectMapping mapping : projectMappings) {
+                logger.debug(mapping.toString());
+            }
```
Comment (bamandel) [inline src/main/java/com/blackducksoftware/integration/jira/config/JiraConfigDeserializer.java]: Is BlackDuckProjectMapping not stringable? Do we specifically want to print separate debugs messages for each of these?

File: src/main/java/com/blackducksoftware/integration/jira/config/JiraConfigDeserializer.java [hunk 2]
```diff
@@ -44,14 +47,23 @@ public BlackDuckJiraConfigSerializable deserializeConfig(final PluginIssueCreati
...
         }
+
         logger.debug("Policy Rules:");
-        for (final PolicyRuleSerializable rule : config.getPolicyRules()) {
-            logger.debug(rule.toString());
+        final List<PolicyRuleSerializable> policyRules = config.getPolicyRules();
+        if (null != policyRules) {
+            for (final PolicyRuleSerializable rule : policyRules) {
+                logger.debug(rule.toString());
+            }
```
Comment (bamandel) [inline src/main/java/com/blackducksoftware/integration/jira/config/JiraConfigDeserializer.java]: Same question as above.

File: src/main/java/com/blackducksoftware/integration/jira/task/BlackDuckMonitor.java [hunk 3]
```diff
@@ -125,7 +125,7 @@ public void reschedule(final long intervalIgnored) {
         }
 
         final HashMap<String, Serializable> blackDuckJobRunnerProperties = new HashMap<>();
-        blackDuckJobRunnerProperties.put(KEY_CONFIGURED_INTERVAL_MINUTES, actualInterval);
+        blackDuckJobRunnerProperties.put(KEY_CONFIGURED_INTERVAL_MINUTES, actualInterval / 1000);
```
Comment (bamandel) [inline src/main/java/com/blackducksoftware/integration/jira/task/BlackDuckMonitor.java]: I would personally prefer this conversion done in a named variable to know what numbers we're dealing with

File: src/main/java/com/blackducksoftware/integration/jira/task/setup/BlackDuckFieldScreenSchemeSetup.java [hunk 4]
```diff
@@ -150,9 +151,14 @@ private OrderableField getOrderedUserFieldFromCustomField(final List<IssueType>
...
-            if (customField == null) {
+            Optional<CustomField> optionalCustomField = jiraServices.getCustomFieldManager()
+                                                            .getCustomFieldObjectsByName(pluginField.getName())
+                                                            .stream()
+                                                            .findFirst();
+            final CustomField customField;
+            if (optionalCustomField.isPresent()) {
+                customField = optionalCustomField.get();
+            } else {
                 customField = createCustomFieldFunction.apply(issueTypeList, pluginField.getName());
```
Comment (bamandel) [inline src/main/java/com/blackducksoftware/integration/jira/task/setup/BlackDuckFieldScreenSchemeSetup.java]: Can this be done in an orElse statement?

**Review by psantos1113 — APPROVED** [review]

File: src/main/java/com/blackducksoftware/integration/jira/task/BlackDuckMonitor.java (hunk 3 above)
Comment (gkillough) [reply src/main/java/com/blackducksoftware/integration/jira/task/BlackDuckMonitor.java]: Oh wow, everything about the interval is very wrong.

**Pull request merged**